doc/source/train/user-guides/checkpoints.rst

```diff
@@ -309,6 +316,77 @@ Lower-performing checkpoints are deleted to save storage space. By default, all
     please ensure that the metric is always reported together with the checkpoints.
 
 
+.. _train-validating-checkpoints:
+
+Validating checkpoints asynchronously
```

Comment:
Recommend reordering this so it's easier to follow conceptually.

1. Introduce the problem (the benefits you've listed)
2. Outline the steps.
     a. Show a simple example here (with just a single model, non-distributed).
     b. The main goal is so that the user understands how `validate_fn`, `validate_config`, `report`, `checkpoint` integrate in their existing training function.
3. Solutions for running distributed (Train, Data)